CMakeLists.txt

```diff
@@ -21,11 +21,9 @@ project(ecdaa
         LANGUAGES C
         VERSION "0.1.1")
 
-add_compile_options(-Werror -Wall -Wextra -std=c99)
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wextra -std=c99")
 
-if (CMAKE_BUILD_TYPE STREQUAL "DebugWithCoverage")
-        SET(CMAKE_C_FLAGS "-g -O0 -fprofile-arcs -ftest-coverage")
-endif()
+SET(CMAKE_C_FLAGS_DEBUGWITHCOVERAGE "${CMAKE_C_FLAGS} -O0 -fprofile-arcs -ftest-coverage")
```

Comment:
```
SET(CMAKE_C_FLAGS_DEBUGWITHCOVERAGE "${CMAKE_C_FLAGS} -O0 -fprofile-arcs -ftest-coverage")
```
to
```
SET(CMAKE_C_FLAGS_DEBUGWITHCOVERAGE "${CMAKE_C_FLAGS_DEBUGWITHCOVERAGE} -O0 -fprofile-arcs -ftest-coverage")
```

Reply:
Does the `SET(CMAKE_C_FLAGS ...` above that just get pulled into every `CMAKE_C_FLAGS*` ?

Reply:
I believe so.

Reply:
Gotcha.

I made that change, and the build looks fine.

Reply:
From the docs, "For example, if CMAKE_BUILD_TYPE is Debug, then CMAKE_C_FLAGS_DEBUG will be added to the CMAKE_C_FLAGS."